File: atcoder/abs/4.c
//usr/bin/env picoc $0 - $@;exit
#include <stdio.h>
int main(){
	int a,b,c,x,i,j,k,r=0;
	scanf("%d%d%d%d",&a,&b,&c,&x);
	for(i=0;i<=x/500;i++)for(j=0;j<=(x-500*i)/100;j++){
		k=x-500*i-100*j;
		r+=k%50==0&&c>=k/50&&a>=i&&b>=j;
	}
	printf("%d\n",r);
	return 0;
}

File: atcoder/abs/4.swift
//usr/bin/env swift $0 $@;exit
let a=Int(readLine()!)!
let b=Int(readLine()!)!
let c=Int(readLine()!)!
let x=Int(readLine()!)!
var r=0
for i in 0...x/500 {
	for j in 0...(x-500*i)/100 {
		let k=x-500*i-100*j
		if k%50==0&&c>=k/50&&a>=i&&b>=j {r+=1}
	}
}
print(r)
